### DISCUSSION TITLE
Question of loop simulation in PWR using THM

### DISCUSSION THREAD

**Discussion by SomnusYu** [idaholab/moose#28116] [Q&A Modules: Thermal Hydraulics] [answered]
Hi all,

I'm simulating a very simple PWR loop using THM. The loop does not contain the heat source. The distribution of pressure and velocity is shown below.
![image](https://github.com/user-attachments/assets/afc6edff-6cd6-4c77-bee5-986d8780b0c7)

![image](https://github.com/user-attachments/assets/5530be5f-28fb-4716-b9c7-f41d828f531f)

The input file is [simple_pwr_loop.txt](https://github.com/user-attachments/files/16186199/simple_pwr_loop.txt).

As in typical PWR, the pressure drop through the pump should be 0.5MPa, which is good in this simulation. But the velocity in the loop should be 5m/s. In this simulation, it is only 0.5 m/s. I do not know the reason.

I tried to change the area of the pipe, but the results did not change.
I also tried to increase the pump head, which could result in an increase in velocity, but following the pressure drop also increased rapidly. 

How to let the pressure drop keep at  0.5MPa, and increase the velocity to 5m/s at the same time?

Thanks!

**Comment by SomnusYu** [comment]
Hi @GiudGiud @joshuahansel,

Could you please kindly give me some suggestions? 

Thanks!

**Comment by GiudGiud** [comment]
> How to let the pressure drop keep at 0.5MPa, and increase the velocity to 5m/s at the same time?

Not sure there is an easy solution there. The pressure drop is tied to the velocity since it partially comes from friction. You could mess with the pipe friction factors, maybe use a different model for the roughness, to reduce pressure drop. The area of the pipe is a variable as well.

Maybe plot the results of pressure_drop as a function of velocity and see if some part of the curve works for you

**Reply by SomnusYu** [reply]
Hi,

I try to let the `f_factor `of all pipe equal to 0, since I just use the simple closures. But nothing changed.
Also, I have tried to change the area of all pipe, it can not change the result. Because the pump area and the pipe area are same.

So, I think both the pipe friction factors and area of pipe are not the key parameters to fix this problem.

I searched some reference: _Andrs D, Berry R, Gaston D, et al. Relap-7 level 2 milestone report: Demonstration of a steady state single phase pwr simulation with relap-7[R]. Idaho National Lab.(INL), Idaho Falls, ID (United States), 2012._
![image](https://github.com/user-attachments/assets/a25f4755-c68b-4a82-8780-dd8eca32310b)
![image](https://github.com/user-attachments/assets/255b61cf-c25e-4882-bc59-3d4d31160996)


The results of RELAP7 could let the pressure drop keep at 0.5MPa, and increase the velocity to 5m/s at the same time. I still have no idea to fix this problem.

**Comment by joshuahansel** [comment]
It appears you are not running close to a steady state. I see that the pressure drops sharply at the junction of your loop with the pressurizer, and this is overshadowing any of the other pressure drops in your system, like by gravity and friction. Running longer decreases this pressure drop. See if you can get to steady state. Perhaps you can try ramping the pump head from zero to your end value to ease the transient.

Ultimately, your pressure drops in your loop need to sum to your pump head. Contributions will be from friction, junctions, and gravity, so your ability to change the steady velocity will be based on the friction factor, since that's the only one you can change.

**Reply by SomnusYu** [reply]
Hi, thanks for your suggestion!

I tried to simulate this case for a long time (5000s) and tried to use a function of the pump head from zero to my needed value. But nothing changed. The pressure drop at the junction connected with the pressurizer is still sharp. Also, the value of the velocity in the loop did not change.

**Reply by joshuahansel** [reply]
I don't know if 5000 s is long enough. I recall seeing the pressure still changing long after that. For example, do something like
```
  end_time = 1000000
  [TimeStepper]
    type = IterationAdaptiveDT
    dt = 10
    optimal_iterations = 5
    iteration_window = 0
    growth_factor = 1.2
    cutback_factor = 0.8
  []
  steady_state_detection = true
```
Another factor here is your junction volumes - try modifying those to see how that changes things.

**Reply by SomnusYu** [reply]
I ran for 100000 seconds now. I found that the velocity can not increase but decreases by about 0.2 m/s. How can the velocity be increased to 5 m/s? The only thing I could change is the pump head to increase the velocity.

> Another factor here is your junction volumes - try modifying those to see how that changes things.

I have tried this way, the result would same after a long time.

**Reply by joshuahansel** [reply]
Ok, I'm thinking that this may be a deficiency in the volume junction formulation. I tried putting another pipe connected to the upper left junction with a dead end (wall) instead of an inlet, and it introduced a similar sharp pressure drop across the junction. I've been struggling all day to try and come up with some improvement to this junction formulation but haven't made progress.

One test you could try would be to put all of your loop pipes in the x-direction and have the pressurizer pipe be orthogonal. The issue appears to be the change of direction coupled with having a connection of more than 2 pipes. Note that the "gravity_vector" parameter can be modified per pipe so that you can get all of the gravity effects correct. Note this is not a long-term solution but just a test.

**Reply by SomnusYu** [reply]
Thank you for your time! 

> One test you could try would be to put all of your loop pipes in the x-direction and have the pressurizer pipe be orthogonal.

Ok, I will have a try. 

By the way, if you figure out this problem in a good way, please do not hesitate to tell me. And I will also try to find the right way and try my best to improve it.

Thanks!!

**Reply by joshuahansel** [reply]
I looked into this more, and on more consideration, I'm not sure that the junction is doing anything wrong. Some kind of pressure drop there is reasonable to me if the pressure losses downstream of the pump cannot match produce the pressurizer's pressure.

Increasing the pump head does in fact increase the velocity in the loop. However, you have an issue with your problem: your setup will just continually heat up your fluid in your loop forever, since you have no heat sink, decreasing the density and thus your pump head will be equivalent to a lower pressure drop over time. As for the pressure drop across your pressurizer junction, I think you're going to see that as long as your pressure losses between pump outlet and pressurizer are not large enough.

**Reply by SomnusYu** [reply]
Thank you for your suggestions and your time!

> your setup will just continually heat up your fluid in your loop forever, since you have no heat sink, decreasing the density and thus your pump head will be equivalent to a lower pressure drop over time. 

I added the heat source of pipe1 (increasing the coolant temperature from 564 to 600 K) and set the `HeatTransferFromSpecifiedTemperature1Phase`(564 K) to pipe3, which is the typical condition in PWR. Now, I can get a steady reasult.
![image](https://github.com/user-attachments/assets/5b0ce9bb-4c25-4e43-bfc3-c601139d038e)

> the pressure losses downstream of the pump cannot match produce the pressurizer's pressure

Then I changed a new friction factor (100000) in pipe1 and pipe2, which is large enough to let the pressure close to the pressure of the pressurize. There are no large pressure drop at the junction of pressurize. That's good.
![image](https://github.com/user-attachments/assets/bcf349d2-9700-480c-ac44-91f6d4c2f177)

However, the velocity decreased to 0.08 m/s since I introduced a lot of frictional drag into the momentum equation in pipe1 and pipe2. 
![image](https://github.com/user-attachments/assets/4b148534-f145-40ac-9df2-c13e5c9ef4a1)

We still have not solved this problem. How do we keep these conditions (pressure and temperature distribution) and just increase the loop velocity to 5m/s? I'm still confused.

Here is the new input file.
[loop.txt](https://github.com/user-attachments/files/16369250/loop.txt)

Thanks!

**Reply by joshuahansel** [reply, marked as answer]
I wouldn't necessarily adjust the friction factor to eliminate that pressure drop at the junction. Just model your system as closely as you're able to your actual reference system:

- Have your input power/heat sink as close as for your BWR case
- Have your friction factor estimated according to some correlation appropriate for your BWR
- Have your piping length system match as well as you can, including the length and flow area
- Increase pump head until you get to 5 m/s

**Reply by SomnusYu** [reply]
Thanks!
